test/end2end_test.go

```diff
@@ -5284,6 +5284,53 @@ func (s) TestGRPCMethod(t *testing.T) {
 	}
 }
 
+// renameProtoCodec is an encoding.Codec wrapper that allows customizing the
+// Name() of another codec.
+type renameProtoCodec struct {
+	encoding.Codec
+	name string
+}
+
+func (r *renameProtoCodec) Name() string { return r.name }
+
+func (s) TestForceCodecName(t *testing.T) {
```

Comment:
Add a top level comment describing this test.

Reply:
Done